src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -220,7 +228,48 @@ void AppHost::_HandleCommandlineArgs()
         // is created.
         if (_windowManager.IsMonarch())
         {
-            _logic.SetNumberOfOpenWindows(_windowManager.GetNumberOfPeasants());
+            const auto numPeasants = _windowManager.GetNumberOfPeasants();
+            const auto layouts = ApplicationState::SharedInstance().PersistedWindowLayouts();
+            if (_logic.ShouldUsePersistedLayout() && layouts && layouts.Size() > 0)
+            {
+                uint32_t startIdx = 0;
+                // If we are the only window, and no commandline arguments were provided
+                // then we should just use the current window to load the first layout
+                if (numPeasants == 1 && !_logic.HasCommandlineArguments())
+                {
+                    _logic.SetPersistedLayoutIdx(startIdx);
+                    startIdx += 1;
+                }

```

Comment:
What about the case where the user has saved windows, and then does something like `wt nt ; sp` - In that case, should we be opening the `nt ; sp` in _this_ window, and then open the persisted state in other windows?

Reply:
Oh wait, I guess it does do that already. That was maybe just a tad bit hard to parse - maybe leave a comment 

```suggestion
                }
                // If we did have a commandline, then we'll open the provided 
                // commandline in this window, and continue on using a new 
                // window for the first persisted layout.
```

Reply:
Updated the comment above this block too